### PR TITLE
Calico - Version update and operator based installation.

### PR DESCRIPTION
**Pull request by kishen-v**
Changes:
1. Update Calico to use the latest release v3.24.5
2. Include option to install Calico using Tigera operator(default).
3. Introduce `installation_type` in `group_vars/all`  to select between operator/manifest based installation.

### CODE REVIEW

File: group_vars/all [hunk 1]
```diff
@@ -56,4 +56,8 @@ build_version: v1.20.0-alpha.0.1402+d39214ade1d60c
 # build_version: d39214ade1d60c
 
 ##### Calico Configurations #####
-calico_version: v3.24.1
+calico_version: v3.24.5
+pod_subnet: 172.20.0.0/16
```
Comment (mkumatag) [inline group_vars/all:60]: lets move this out of calico section

File: group_vars/all [hunk 2]
```diff
@@ -56,4 +56,8 @@ build_version: v1.20.0-alpha.0.1402+d39214ade1d60c
...
+# The available methods of installation for Calico can be either using the tigera operator, or using the manifest.
+# Choose between "operator" (default) or "manifest" for the desired installation method.
+installation_type: operator
```
Comment (mkumatag) [inline group_vars/all:63]: - lets prefix with calico
- lets have manifest as a default for now, will migrate to operator later

File: roles/install-calico/tasks/calico-operator.yaml [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+    url: "https://raw.githubusercontent.com/projectcalico/calico/{{ calico_version }}/manifests/custom-resources.yaml"
+    dest: /tmp/calico.yaml
+    mode: '0755'
+
+# Guide to set v-eth mtu for calico - https://projectcalico.docs.tigera.io/networking/mtu
+- name: Set veth_mtu to {{ calico_mtu }}
+  lineinfile:
+    path: /tmp/calico.yaml
+    insertafter: "calicoNetwork:"
+    line: "    mtu: {{ calico_mtu }}"
```
Comment (mkumatag) [inline roles/install-calico/tasks/calico-operator.yaml:17]: lets add a TODO to have a better way to add a field into yaml spec

File: roles/install-calico/tasks/calico-operator.yaml [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+    path: /tmp/calico.yaml
+    regexp: "cidr:.*"
+    replace: "cidr: {{ pod_subnet }}"
+
+- name: Set up Calico CNI from manifest
+  command: kubectl create -f /tmp/calico.yaml
+
+# TODO: Switch to suitable logic when - https://github.com/kubernetes/kubernetes/issues/83242 is fixed.
+- name: Wait until namespace is created for calico-apiserver.
+  shell: while ! kubectl get namespace calico-apiserver; do sleep 5; done
```
Comment (mkumatag) [inline roles/install-calico/tasks/calico-operator.yaml:30]: lets see if we can adopt this:
```yaml
- name: Wait until namespace is created for calico-apiserver with timeout
  command: kubectl get namespace calico-apiserver
  register: result
  until: result.stdout.find("Active") != -1
  retries: 12
  delay: 5
```

**Review by mkumatag — CHANGES_REQUESTED** [review]

File: group_vars/all (hunk 1 above)
Comment (kishen-v) [reply group_vars/all]: Moved `pod_subnet` field to Kubernetes configuration section of the file.

File: group_vars/all (hunk 2 above)
Comment (kishen-v) [reply group_vars/all]: Addressed.

File: roles/install-calico/tasks/calico-operator.yaml (hunk 3 above)
Comment (kishen-v) [reply roles/install-calico/tasks/calico-operator.yaml:18]: Addressed.

File: roles/install-calico/tasks/calico-operator.yaml (hunk 4 above)
Comment (kishen-v) [reply roles/install-calico/tasks/calico-operator.yaml]: Applied suggestion. Works as intended with a timeout of 90s. (5s * 18 retries.)

**Pull request merged**